samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+};
+
+/* Specifying the expected options */
+static struct option long_options[] = {
+	{ "method", required_argument, 0, 'm' },
+	{ "interval", required_argument, 0, LOCATION_SHELL_OPT_INTERVAL },
+	{ "gnss_accuracy", required_argument, 0, LOCATION_SHELL_OPT_GNSS_ACCURACY },
+	{ "gnss_timeout", required_argument, 0, LOCATION_SHELL_OPT_GNSS_TIMEOUT },
+	{ "cellular_timeout", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_TIMEOUT },
+	{ "cellular_service", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_SERVICE },
+	{ "wifi_timeout", required_argument, 0, LOCATION_SHELL_OPT_WIFI_TIMEOUT },
+	{ "wifi_service", required_argument, 0, LOCATION_SHELL_OPT_WIFI_SERVICE },
+	{ 0, 0, 0, 0 }
+};
+
+/******************************************************************************/
+
+static void location_shell_print_usage(const struct shell *shell,
+				       struct location_shell_cmd_args *loc_cmd_args)
+{
+	switch (loc_cmd_args->command) {
+	case LOCATION_CMD_GET:
+		mosh_print_no_format(location_get_usage_str);
+		break;
+
+	default:
+		mosh_print_no_format(location_usage_str);
+		break;
+	}
+}
+
+#define MOSH_LOC_SERVICE_NONE 0xFF
+
+static enum location_service location_shell_string_to_service(const char *service_str)
+{
+	enum location_service service = MOSH_LOC_SERVICE_NONE;
+
+	if (strcmp(service_str, "any") == 0) {
+		service = LOCATION_SERVICE_ANY;
+	} else if (strcmp(service_str, "nrf") == 0) {
+		service = LOCATION_SERVICE_NRF_CLOUD;
+	} else if (strcmp(service_str, "here") == 0) {
+		service = LOCATION_SERVICE_HERE;
+	} else if (strcmp(service_str, "skyhook") == 0) {
+		service = LOCATION_SERVICE_SKYHOOK;
+	} else if (strcmp(service_str, "polte") == 0) {
+		service = LOCATION_SERVICE_POLTE;
+	}
+
+	return service;
+}
+
+/******************************************************************************/
+
+void location_ctrl_event_handler(const struct location_event_data *event_data)
+{
+	char gmaps_str[512];
+
+	switch (event_data->id) {
+	case LOCATION_EVT_LOCATION:
+		mosh_print("Location:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->location.method),
+			event_data->location.method);
+		mosh_print("  latitude: %.06f", event_data->location.latitude);
+		mosh_print("  longitude: %.06f", event_data->location.longitude);
+		mosh_print("  accuracy: %.01f m", event_data->location.accuracy);
+		if (event_data->location.datetime.valid) {
+			mosh_print(
+				"  date: %04d-%02d-%02d",
+				event_data->location.datetime.year,
+				event_data->location.datetime.month,
+				event_data->location.datetime.day);
+			mosh_print(
+				"  time: %02d:%02d:%02d.%03d UTC",
+				event_data->location.datetime.hour,
+				event_data->location.datetime.minute,
+				event_data->location.datetime.second,
+				event_data->location.datetime.ms);
+		}
+		sprintf(gmaps_str, "  Google maps URL: https://maps.google.com/?q=%f,%f",
+			event_data->location.latitude, event_data->location.longitude);
+		mosh_print("%s", gmaps_str);
```

Comment:
I guess we could try to remove `gmaps_str` completely and add the format string to `mosh_print` directly. If it won't work, we could do like in Location library function `location_core_event_cb()`, i.e., convert floats to strings first with small 12 character string buffers. Anyway, `gmaps_str` should be made a lot smaller if it remains.

Reply:
This seems to work:
```
		mosh_print(
			"  Google maps URL: https://maps.google.com/?q=%f,%f",
			event_data->location.latitude, event_data->location.longitude);
```